### PR TITLE
feat rag with langchain

### PR DESCRIPTION
**Pull request by loks0n**

### CODE REVIEW

File: node/prompt-chatgpt/src/main.js [hunk 1]
```diff
@@ -16,11 +16,7 @@ export default async ({ req, res }) => {
     return res.json({ ok: false, error: err.message }, 400);
   }
 
-  const openai = new OpenAIApi(
-    new Configuration({
-      apiKey: process.env.OPENAI_API_KEY,
-    })
-  );
+  const openai = new OpenAIApi();
```
Comment (christyjacob4) [inline node/prompt-chatgpt/src/main.js:19]: How do we pass the API Key if this is removed

File: node/sync-with-pinecone/src/main.js [hunk 2]
```diff
@@ -32,9 +30,7 @@ export default async ({ req, res, log }) => {
 
   const databases = new Databases(client);
 
-  const openai = new OpenAI({
-    apiKey: process.env.OPENAI_API_KEY,
-  });
+  const openai = new OpenAI();
```
Comment (christyjacob4) [inline node/sync-with-pinecone/src/main.js:33]: is the API key auto detected?

File: node/rag-with-langchain/README.md [hunk 3]
```diff
@@ -0,0 +1,139 @@
...
+
+A retrieval augmented generation (RAG) model for completions with the LangChain API, which provides a simple interface for querying the model. An endpoint is provided for indexing data from an Appwrite collection to a Pinecone vector database.
+
+## 🧰 Usage
+
+### GET /
+
+HTML form for interacting with the function.
+
+### POST /
```
Comment (christyjacob4) [inline node/rag-with-langchain/README.md:11]: do you think we should be more verbose and call this POST /index?

File: node/text-to-speech-with-huggingface/src/main.js [hunk 4]
```diff
@@ -0,0 +1,47 @@
+import { HfInference } from '@huggingface/inference';
+import { getStaticFile, throwIfMissing } from './utils.js';
+
+export default async ({ req, res }) => {
+  throwIfMissing(process.env, ['HUGGINGFACE_ACCESS_TOKEN']);
```
Comment (christyjacob4) [inline node/text-to-speech-with-huggingface/src/main.js:5]: Is this ACCESS_TOKEN or API_TOKEN? In the readme for this function I see API_TOKEN. 
Please search through the other examples as well to make sure we use the same variable

File: node/text-to-speech-with-huggingface/src/main.js [hunk 5]
```diff
@@ -0,0 +1,47 @@
...
+  }
+
+  if (!req.body.source || typeof req.body.source !== 'string') {
+    return res.json(
+      { ok: false, error: 'Missing required field `source`' },
+      400
+    );
+  }
+
+  const hf = new HfInference(process.env.HUGGINGFACE_ACCESS_TOKEN);
```
Comment (christyjacob4) [inline node/text-to-speech-with-huggingface/src/main.js:24]: Is this ACCESS_TOKEN or API_TOKEN? In the readme for this function I see API_TOKEN. 
Please search through the other examples as well to make sure we use the same variable

File: node/text-to-speech-with-huggingface/README.md [hunk 6]
```diff
@@ -95,12 +95,12 @@ The ID of the collection where the responses are stored.
 | Required     | No               |
 | Sample Value | `text_to_speech` |
 
-### HUGGING_FACE_API_KEY
+### HUGGINGFACE_API_TOKEN
```
Comment (christyjacob4) [inline node/text-to-speech-with-huggingface/README.md]: In the function, this is called `HUGGINGFACE_ACCESS_TOKEN`

File: node/speech-recognition-with-huggingface/src/main.js [hunk 7]
```diff
@@ -1,73 +1,47 @@
...
-import AppwriteService from './appwrite.js';
+import { getStaticFile, throwIfMissing } from './utils.js';
 
-export default async ({ req, res, log, error }) => {
-  throwIfMissing(process.env, [
-    'HUGGING_FACE_API_KEY',
-    'APPWRITE_API_KEY',
-  ]);
+export default async ({ req, res }) => {
+  throwIfMissing(process.env, ['HUGGINGFACE_ACCESS_TOKEN']);
```
Comment (christyjacob4) [inline node/speech-recognition-with-huggingface/src/main.js:5]: In the readme this is called `HUGGINGFACE_API_TOKEN`

File: node/sync-with-pinecone/src/main.js (hunk 2 above)
Comment (loks0n) [reply node/sync-with-pinecone/src/main.js:33]: Yeah

File: node/prompt-chatgpt/src/main.js (hunk 1 above)
Comment (loks0n) [reply node/prompt-chatgpt/src/main.js:19]: It's auto detected

File: node/text-to-speech-with-huggingface/src/main.js (hunk 4 above)
Comment (loks0n) [reply node/text-to-speech-with-huggingface/src/main.js:5]: I've standardised this to ACCESS_TOKEN - this is how Hugging Face refer it on their docs

File: node/text-to-speech-with-huggingface/src/main.js (hunk 5 above)
Comment (loks0n) [reply node/text-to-speech-with-huggingface/src/main.js:24]: ditto

File: node/text-to-speech-with-huggingface/README.md (hunk 6 above)
Comment (loks0n) [reply node/text-to-speech-with-huggingface/README.md]: fixed

File: node/speech-recognition-with-huggingface/src/main.js (hunk 7 above)
Comment (loks0n) [reply node/speech-recognition-with-huggingface/src/main.js]: fixed the readme

**Pull request merged**